pkg/storage/pvc.go

```diff
@@ -20,9 +28,151 @@ type PVCBuilder struct {
 	// Created persistentvolumeclaim object
 	Object *v1.PersistentVolumeClaim
 
+	errorMsg  string
 	apiClient *clients.Settings
 }
 
+// NewPVCBuilder creates a new structure for persistentvolumeclaim.
+func NewPVCBuilder(apiClient *clients.Settings, name, nsname string) *PVCBuilder {
+	glog.V(100).Infof("Creating PersistentVolumeClaim %s in namespace %s",
+		name, nsname)
+
+	builder := PVCBuilder{
+		Definition: &v1.PersistentVolumeClaim{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+			Spec: v1.PersistentVolumeClaimSpec{},
+		},
+	}
+
+	builder.apiClient = apiClient
+
+	if name == "" {
+		glog.V(100).Infof("PVC name is emtpty")
+
+		builder.errorMsg = "PVC name is empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("PVC namespace is emtpty")
+
+		builder.errorMsg = "PVC namespace is empty"
+	}
+
+	return &builder
+}
+
+// WithPVCAccessMode configure access mode for the PV.
+func (builder *PVCBuilder) WithPVCAccessMode(accessMode string) (*PVCBuilder, error) {
+	glog.V(100).Infof("Set PVC accessMode: %s", accessMode)
+
+	if accessMode == "" {
+		glog.V(100).Infof("Empty accessMode for PVC %s", builder.Definition.Name)
+		builder.errorMsg = "Empty accessMode for PVC requested"
+
+		return builder, fmt.Errorf(builder.errorMsg)
+	}
+
+	if !validatePVCAccessMode(accessMode) {
+		glog.V(100).Infof("Invalid accessMode for PVC %s", accessMode)
+		builder.errorMsg = fmt.Sprintf("Invalid accessMode for PVC %s", accessMode)
+
+		return builder, fmt.Errorf(builder.errorMsg)
+	}
+
+	if builder.Definition.Spec.AccessModes != nil {
+		builder.Definition.Spec.AccessModes = append(builder.Definition.Spec.AccessModes,
+			v1.PersistentVolumeAccessMode(accessMode))
+	} else {
+		builder.Definition.Spec.AccessModes = []v1.PersistentVolumeAccessMode{v1.PersistentVolumeAccessMode(accessMode)}
+	}
+
+	return builder, nil
+}
+
+// validatePVCAccessMode validates if requested mode is valid for PVC.
+func validatePVCAccessMode(accessMode string) bool {
+	glog.V(100).Info(fmt.Sprintf("Validating accessMode %s", accessMode))
```

Comment:
```suggestion
	glog.V(100).Infof("Validating accessMode %s", accessMode)
```

Reply:
@cdvultur suggestion doesn't work as `%s` is not translated but the `.Info()` omitting it works though

Reply:
updated